src/Sulu/Bundle/AdminBundle/FormMetadata/FormMetadataMapper.php

```diff
@@ -203,6 +206,24 @@ private function mapSchemaProperties(array $itemsMetadata): array
                 return $this->mapSchemaProperties($itemMetadata->getChildren());
             }
 
+            if ($itemMetadata instanceof ContentBlockMetadata) {
+                $blockTypeSchemas = [];
+                foreach ($itemMetadata->getComponents() as $blockType) {
+                    $blockTypeSchemas[] = new SchemaMetadata(
+                        array_merge(
+                            $this->mapSchemaProperties($blockType->getChildren()),
+                            ['type' => new ConstMetadata('type', true, $blockType->getName())]
+                        )
+                    );
+                }
+
+                return new ArrayMetadata(
+                    $itemMetadata->getName(),
+                    $itemMetadata->isRequired(),
+                    new SchemaMetadata([], $blockTypeSchemas)
+                );
+            }
+
             if (!$itemMetadata->isRequired()) {
```

Comment:
does it still make sense to return here? 🤔

Reply:
No, I think that is not necessary anymore :slightly_smiling_face: